strawberry/alloptional.py

```diff
@@ -0,0 +1,23 @@
+from abc import ABCMeta
+from typing import Optional
+
+import strawberry
+
+
+class AllOptional(ABCMeta):
+    def __new__(cls, name: str, bases: tuple, namespaces: dict, **kwargs):
+        annotations = namespaces.get("__annotations__", {})
+        fields: list[str] = []
+        for base in bases:
+            annotations.update(base.__annotations__)
```

Comment:
I think this would need to use the MRO so that all the classes in the inheritance hierarchy are considered, and in the correct order.

The problem here is that, since the class has not been created yet, you can't access `klass.__mro__` yet. Maybe there's a way to construct that MRO given the `bases`?

Reply:
Exactly! Just remember to iterate over them in the reverse order, or else lower priority bases (the ones that come later in the MRO) will override the annotations from the higher priority ones